openapi/openapi.yaml

```diff
@@ -247,3 +254,202 @@ paths:
+      type: string
+      maxLength: 64
+      example: 'my_namespace'
+      pattern: '^(?!.*__)[a-z]+[0-9a-z_]*$'
+
+    NamespacesPage:
+      description: "Paginated list of Namespaces"
+      allOf:
+        - $ref: '#/components/schemas/PageInfo'
+        - type: object
+          properties:
+            data:
+              description: 'List of Namespaces for this page'
+              title: 'Namespaces'
+              type: array
+              items:
+                $ref: '#/components/schemas/Namespace'
+          required:
+            - data
+
+    PageCount:
+      description: 'The number of items in this Page'
+      type: integer
+      example: 5
+
+    PageInfo:
+      description: 'Pagination info'
+      title: 'Page Info'
+      properties:
+        meta:
+          $ref: '#/components/schemas/PageMeta'
+        links:
+          $ref: '#/components/schemas/PageLinks'
+      required:
+        - meta
+        - links
+
+    PageLinks:
+      type: object
+      title: 'PageLinks'
+      properties:
+        first:
+          type: string
+          description: 'Link to first page'
+          format: uri
+          example: '/namespaces/?offset=0&limit=1'
+        last:
+          type: string
+          description: 'Link to last page'
+          format: uri
+          example: '/namespaces/?offset=7&limit=1'
+        next:
+          type: string
+          description: 'Link to next page'
+          format: uri
+          example: '/namespaces/?offset=5&limit=1'
+          nullable: true
+        previous:
+          type: string
+          description: 'Link to next page'
+          format: uri
+          example: '/namespaces/?offset=3&limit=1'
+          nullable: true
+      required:
+        - first
+        - last
+        - next
+        - previous
+
+    PageMeta:
+      title: 'PageMeta'
+      type: object
+      properties:
+        count:
+          $ref: '#/components/schemas/PageCount'
+      required:
+        - count
+
+    User:
+      title: 'User'
+      description: 'Automation Hub User'
+      type: object
+
+  parameters:
+
+    PageLimit:
+      description: 'Number of results to return per page.'
+      in: query
+      name: limit
+      required: false
+      schema:
+        type: integer
+        default: 10
+        minimum: 1
+        maximum: 100
+
+    PageOffset:
+      description: 'Page offset number within the paginated result set'
+      in: query
+      name: offset
+      required: false
+      schema:
+        type: integer
+        default: 0
+        minimum: 0
+
+    NamespaceName:
+      description: 'Namespace name'
+      in: path
+      name: name
+      required: true
+      schema:
+        type: string
+
+  requestBodies:
+
+    Namespace:
+      description: "A Namespace body"
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/Namespace'
+
+  responses:
+
+    DeleteSuccessful:
```

Comment:
Why do we need this one?

Reply:
It's reused by other DELETE entrypoints (when they are added in later revisions)

Reply:
Which means having the same description for all DELETE operations. What's the point?

Please remove this and rollback to original solution.

Reply:
updated